src/instructlab/cli/model/upload.py

```diff
@@ -0,0 +1,245 @@
+            )
+
+        click.echo(
+            f"Uploading model to Hugging Face:\n{DEFAULT_INDENT}Model: {self.model}\n{DEFAULT_INDENT}Destination: {self.upload_dest}@{self.release}\n"
+        )
+
+        # set logger
+        if self.ctx.obj is not None:
+            hf_logging.set_verbosity(self.ctx.obj.config.general.log_level.upper())
+
+        # determine if model is path or name
+        if os.path.exists(self.model):
+            self.local_model_path = Path(self.model)
+            self.model = self.model.split("/")[-1]
+        else:
+            self.local_model_path = os.path.join(self.model_dir, self.model)
+            # throw an error if the built path is invalid
+            if not os.path.exists(self.local_model_path):
+                click.secho(
+                    f"Couldn't find model at {self.local_model_path} - are you sure it exists?",
+                    fg="red",
+                )
+                raise click.exceptions.Exit(1)
+
+        # upload either safetensors or gguf, if valid
+        if is_model_safetensors(self.local_model_path):
+            self.upload_safetensors()
+        elif is_model_gguf(self.local_model_path):
+            self.upload_gguf()
+        else:
+            click.secho(
+                f"Local model path {self.local_model_path} is a valid path, but is neither safetensors nor a GGUF - cannot upload to Hugging Face",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+
+    def upload_gguf(self) -> None:
+        try:
+            hf_api = HfApi()
+            hf_api.upload_file(
+                path_or_fileobj=self.local_model_path,
+                path_in_repo=self.model,
+                repo_id=self.upload_dest,
+                revision=self.release,
+                token=self.hf_token,
+            )
+        except Exception as exc:
+            click.secho(
+                f"\nUploading GGUF model failed with the following Hugging Face Hub error:\n{DEFAULT_INDENT}{exc}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        click.secho(
+            "\nUploading GGUF model succeeded!",
+            fg="green",
+        )
+
+    def upload_safetensors(self) -> None:
+        try:
+            hf_api = HfApi()
+            hf_api.upload_folder(
+                folder_path=self.local_model_path,
+                repo_id=self.upload_dest,
+                revision=self.release,
+                token=self.hf_token,
+            )
+        except Exception as exc:
+            click.secho(
+                f"\nUploading safetensors model failed with the following Hugging Face Hub error:\n{DEFAULT_INDENT}{exc}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        click.secho(
+            "\nUploading safetensors model succeeded!",
+            fg="green",
+        )
```

Comment:
Same as above, but for `upload_folder`: https://github.com/huggingface/huggingface_hub/blob/v0.26.5/src/huggingface_hub/hf_api.py#L4971L4878

```suggestion
    def upload_safetensors(self) -> None:
        try:
            hf_api = HfApi()
            resp = hf_api.upload_folder(
                folder_path=self.local_model_path,
                repo_id=self.upload_dest,
                revision=self.release,
                token=self.hf_token,
            )
        except Exception as exc:
            click.secho(
                f"\nUploading safetensors model failed with the following Hugging Face Hub error:\n{DEFAULT_INDENT}{exc}",
                fg="red",
            )
            raise click.exceptions.Exit(1)
        click.secho(
            f"\nUploading safetensors model succeeded!  Hugging Face response:\n{resp}",
            fg="green",
        )
```